python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -391,13 +388,12 @@ def _cache_actions(self, data, config_data):
         arg_config_data = dict(
             lookup_hash = config_data["lookup_hash"],
             contents_hash=config_data["contents_hash"],
-            entity=copy.copy(config_data["entity"]),
+            entity=config_data["entity"],
         )
 
         # If we have a descriptor, we don't want to pickle that. It's not
         # needed in either the caching or execution scripts.
-        if "descriptor" in arg_config_data["entity"]:
-            del arg_config_data["entity"]["descriptor"]
+        del arg_config_data["entity"]["descriptor"]
```

Comment:
If the descriptor is always present, then you can probably do away with the "If we have a descriptor" bit.